app/src/main/java/com/ncnf/user/User.java

```diff
@@ -1,203 +1,240 @@
+    public void setSavedEventsIds(List<String> savedEventsIds) {
+        this.savedEventsIds = savedEventsIds;
     }
 
-    public CompletableFuture<DatabaseResponse>  updateLastName(String name){
-        return this.update(LAST_NAME_KEY, name);
+    public void setBirthDate(Date birthDate) {
+        this.birthDate = birthDate;
     }
 
-    public CompletableFuture<DatabaseResponse>  updateFirstName(String name){
-        return this.update(FIRST_NAME_KEY, name);
+    public void setNotifications(boolean notifications) {
+        this.notifications = notifications;
     }
 
-    public CompletableFuture<DatabaseResponse>  updateBirth(int year){
-        return this.update(BIRTH_YEAR_KEY, year);
+    public CompletableFuture<Boolean>  updateNotifications(boolean isEnabled) {
+        setNotifications(isEnabled);
+        return this.db.updateField(USERS_COLLECTION_KEY+uuid, NOTIFICATIONS_KEY, isEnabled);
     }
 
-    public CompletableFuture<DatabaseResponse>  updateNotifications(boolean isEnabled) {
-        return this.update(NOTIFICATIONS_KEY, isEnabled);
+    public CompletableFuture<Boolean>  updateNotificationsToken(String token) {
+        return this.db.updateField(USERS_COLLECTION_KEY+uuid, NOTIFICATIONS_TOKEN_KEY, token);
     }
 
-    public CompletableFuture<DatabaseResponse>  updateNotificationsToken(String token) {
-        return this.update(NOTIFICATIONS_TOKEN_KEY, token);
+    public CompletableFuture<Boolean> saveUserToDB(){
+        return this.db.setDocument(USERS_COLLECTION_KEY + uuid, this);
     }
 
-    public CompletableFuture<DatabaseResponse> loadUserFromDB(){
-        return this.db.getData(this.path);
+    public CompletableFuture<User> loadUserFromDB(){
+        CompletableFuture<User> futureUser = this.db.getDocument(USERS_COLLECTION_KEY + uuid, User.class);
+
+        return futureUser.thenApply(response -> {
+            User user = response;
+
+            this.username = user.getUsername();
+            this.email = user.getEmail();
+            this.firstName = user.getFirstName();
+            this.lastName = user.getLastName();
+            this.friendsIds = user.getFriendsIds();
+            this.ownedEventsIds = user.getOwnedEventsIds();
+            this.savedEventsIds = user.getSavedEventsIds();
+            this.birthDate = user.getBirthDate();
+            this.notifications = user.getNotifications();
+
+            return this;
+        }).exceptionally(exception -> {
+            // TODO : handle exception
+            return null;
+        });
     }
 
-    public CompletableFuture<DatabaseResponse> saveEvent(Event event){
-        return this.addEvent(event, SAVED_EVENTS_KEY);
+
+    public CompletableFuture<List<User>> getFriends(){
+        if(this.friendsIds.isEmpty()){
+            return CompletableFuture.completedFuture(new ArrayList<>());
+        }
+
+        return this.db.whereIn(USERS_COLLECTION_KEY, UUID_KEY, this.friendsIds, User.class);
```

Comment:
As discussed, wouldn't the query be more efficient by searching all users which have this.uuid in their friendsIds list (with query ArrayContains) ?

Reply:
This was a test to see if everything worked correctly. But we'll discuss some changes in the User class during our meeting today.